backend/ctgov/api/views.py

```diff
@@ -130,40 +160,184 @@ def post(self, request, *args, **kwargs):
+        elif valid_number(eligibility_min_age):
+            q_age_between = Q(
+                eligibility_min_age_numeric__gte=convert_to_number(
+                    eligibility_min_age
+                )
+            )
+        elif valid_number(eligibility_max_age):
+            q_age_between = Q(
+                eligibility_max_age_numeric__lte=convert_to_number(
+                    eligibility_max_age
+                )
+            )
+        else:
+            pass
+
         if not first:
             first = 0
         if not last:
             last = 100
 
-        search_results = (
-            BasicSearchM.objects.filter(**filters)
-            .all()
-            .values(
+        if metadata_required:
+            search_results_all = (
+                SearchStudies.objects.filter(**filters)
+                .filter(q_title_acronym)
+                .filter(q_outcome_measure)
+                .filter(q_eligibility_age)
+                .filter(q_age_between)
+                .all()
+            )
+
+            search_results = search_results_all.all().values(
                 "status",
                 "brief_title",
                 "nct_id",
                 "condition_name",
                 "intervention_name",
                 "location_name",
             )[first:last]
-        )
 
-        serializer = BasicSearchSerializer(search_results, many=True)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+            results_count = search_results_all.count()
+
+        else:
+            search_results = (
+                SearchStudies.objects.filter(**filters)
+                .filter(q_title_acronym)
+                .filter(q_outcome_measure)
+                .filter(q_eligibility_age)
+                .filter(q_age_between)
+                .all()
+                .values(
+                    "status",
+                    "brief_title",
+                    "nct_id",
+                    "condition_name",
+                    "intervention_name",
+                    "location_name",
+                )[first:last]
+            )
+
+            results_count = "NA"
+
+        serializer = SearchStudiesSerializer(search_results, many=True)
+        return Response(
+            {
+                "metadata": [{"results_count": str(results_count)}],
+                "search_results": serializer.data,
+            },
+            status=status.HTTP_200_OK,
+        )
+        # return Response(serializer.data, status=status.HTTP_200_OK)
 
 
+# Convert passed date parameter to match with db format
 def convert_to_date(datestr):
     date = datetime.strptime(datestr, "%Y-%m-%d")
     return date
 
 
+# Check whether passed date parameter is valid
 def valid_date(datestr):
-    if not datestr:
-        return False
+    status = True
+    try:
+        if datestr:
+            datetime.strptime(str(datestr), "%Y-%m-%d")
+        else:
+            status = False
+    except (ValueError, NameError):
```

Comment:
Remove NameError here.

Check for null case outside of try/except and return false directly rather than saving it to a local variable.

Get rid of status variable. Just return true or false.

Check the type of datestr outside of try/except, rather than converting to a string.

Reply:
I don't think i agree with this.